x-pack/test/apm_api_integration/tests/alerts/anomaly_alert.spec.ts

```diff
@@ -0,0 +1,143 @@
+        const serviceA = apm.service('service-a', 'production', 'java').instance('a');
+
+        const events = timerange(new Date(start).getTime(), new Date(end).getTime())
+          .interval('1m')
+          .rate(1)
+          .spans((timestamp) => {
+            const isInSpike = timestamp >= spikeStart && timestamp < spikeEnd;
+            const count = isInSpike ? 4 : NORMAL_RATE;
+            const duration = isInSpike ? 1000 : NORMAL_DURATION;
+            const outcome = isInSpike ? 'failure' : 'success';
+
+            return [
+              ...range(0, count).flatMap((_) =>
+                serviceA
+                  .transaction('tx', 'request')
+                  .timestamp(timestamp)
+                  .duration(duration)
+                  .outcome(outcome)
+                  .serialize()
+              ),
+            ];
+          });
+
+        await synthtraceEsClient.index(events);
+        await createAndRunApmMlJob({ environment: 'production', ml });
+        const { body: createdRule } = await supertest
+          .post(`/api/alerting/rule`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            params: {
+              environment: 'production',
+              serviceName: 'service-a',
+              transactionType: 'request',
+              windowSize: 30,
+              windowUnit: 'm',
+              anomalySeverityType: 'warning',
+            },
+            consumer: 'apm',
+            schedule: {
+              interval: '1m',
+            },
+            tags: ['apm', 'service.name:service-a'],
+            name: 'Latency anomaly | service-a',
+            rule_type_id: AlertType.Anomaly,
+            notify_when: 'onActiveAlert',
+            actions: [],
+          });
+        ruleId = createdRule.id;
+      });
+
+      after(async () => {
+        await synthtraceEsClient.clean();
+        await ml.cleanMlIndices();
+        await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'foo');
+      });
+
+      const WaitForStatusIncrement = 500;
+
+      async function waitForStatus({
+        id,
+        statuses,
+        waitMillis = 10000,
+      }: {
+        statuses: Set<string>;
+        waitMillis?: number;
+        id?: string;
+      }): Promise<Record<string, any>> {
+        if (waitMillis < 0 || !id) {
+          expect().fail(`waiting for alert ${id} statuses ${Array.from(statuses)} timed out`);
+        }
+
+        const response = await supertest.get(`/api/alerting/rule/${id}`);
+        expect(response.status).to.eql(200);
+
+        const { execution_status: executionStatus } = response.body || {};
+        const { status } = executionStatus || {};
+
+        const message = `waitForStatus(${Array.from(statuses)}): got ${JSON.stringify(
+          executionStatus
+        )}`;
+
+        if (statuses.has(status)) {
+          return executionStatus;
+        }
+
+        // eslint-disable-next-line no-console
+        console.log(`${message}, retrying`);
```

Comment:
we have a logging service for this purpose, we can use the debug() method there and have this not pollute test run output.